gitoxide-core/src/repository/log.rs

```diff
@@ -0,0 +1,152 @@
+use gix::bstr::{BStr, BString, ByteSlice};
+use gix::prelude::FindExt;
+use gix::ObjectId;
+
+pub fn log(mut repo: gix::Repository, out: &mut dyn std::io::Write, pathspec: BString) -> anyhow::Result<()> {
+    repo.object_cache_size_if_unset(repo.compute_object_cache_size_for_tree_diffs(&**repo.index_or_empty()?));
+
+    let head = repo.head()?.peel_to_commit_in_place()?;
+    let infos: Vec<_> =
+        gix::traverse::commit::topo::Builder::from_iters(&repo.objects, [head.id], None::<Vec<gix::ObjectId>>)
+            .build()?
+            .collect();
+
+    let infos: Vec<_> = infos
+        .into_iter()
+        .filter(|info| {
+            let commit = repo.find_commit(info.as_ref().unwrap().id).unwrap();
```

Comment:
The excessive`.unwrap()` seems to be a sign that the `filter` closure is too limiting.